src/borg/helpers.py

```diff
@@ -773,6 +773,14 @@ def parse(self, text):
         if not valid:
             return False
         self.archive = m.group('archive')
+        return self._is_archive_name_printable()
+
+    # lookout for non-printable characters
+    def _is_archive_name_printable(self):
+        if self.archive:
+            for char in self.archive:
+                if unicodedata.category(char)[0] == 'C':
+                    return False
         return True
```

Comment:
I've looked it up, such a function, dissapointedly, doesn't exist. The closest functions would be string.printable and curses.ascii.isprint(), but they only only on ASCII, not unicode.